src/mpc/cli/mpc_phase2_verify_transcript.cpp

```diff
@@ -0,0 +1,217 @@
+    std::string digest;
+
+public:
+    mpc_phase2_verify_transcript()
+        : subcommand("phase2-verify-transcript")
+        , challenge_0_file()
+        , transcript_file()
+        , final_challenge_file()
+        , digest()
+    {
+    }
+
+private:
+    void initialize_suboptions(
+        po::options_description &options,
+        po::options_description &all_options,
+        po::positional_options_description &pos) override
+    {
+        options.add_options()(
+            "digest",
+            po::value<std::string>(),
+            "Check that transcript includes contribution digest");
+        all_options.add(options).add_options()(
+            "challenge_0_file", po::value<std::string>(), "challenge file")(
+            "transcript_file", po::value<std::string>(), "transcript file")(
+            "final_challenge_file",
+            po::value<std::string>(),
+            "final challenge file");
+        pos.add("challenge_0_file", 1)
+            .add("transcript_file", 1)
+            .add("final_challenge_file", 1);
+    }
+
+    void parse_suboptions(const po::variables_map &vm) override
+    {
+        if (0 == vm.count("challenge_0_file")) {
+            throw po::error("challenge_0_file not specified");
+        }
+        if (0 == vm.count("transcript_file")) {
+            throw po::error("transcript_file not specified");
+        }
+        if (0 == vm.count("final_challenge_file")) {
+            throw po::error("final_challenge_file not specified");
+        }
+        challenge_0_file = vm["challenge_0_file"].as<std::string>();
+        transcript_file = vm["transcript_file"].as<std::string>();
+        final_challenge_file = vm["final_challenge_file"].as<std::string>();
+        digest = vm.count("digest") ? vm["digest"].as<std::string>() : "";
+    }
+
+    void subcommand_usage() override
+    {
+        std::cout << "Usage:\n  " << subcommand_name
+                  << " \\\n    <challenge_0_file> <transcript_file> "
+                     "<final_challenge_file>\n\n";
+    }
+
+    int execute_subcommand() override
+    {
+        if (verbose) {
+            std::cout << "challenge_0: " << challenge_0_file << "\n"
+                      << "transcript: " << transcript_file << "\n"
+                      << "final_challenge: " << final_challenge_file
+                      << std::endl;
+        }
+
+        // Load the initial challenge
+        libff::enter_block("Load challenge_0 file");
+        const srs_mpc_phase2_challenge<ppT> challenge_0 = [&]() {
+            std::ifstream in(
+                challenge_0_file, std::ios_base::binary | std::ios_base::in);
+            in.exceptions(
+                std::ios_base::eofbit | std::ios_base::badbit |
+                std::ios_base::failbit);
+            return srs_mpc_phase2_challenge<ppT>::read(in);
+        }();
+        libff::leave_block("Load challenge_0 file");
+
+        // Simple sanity check on challenge.0.  The initial transcript digest
```

Comment:
`challenge.0.  The` -> double spacing. Please use a single space after `.`

Reply:
Removed